yt/fields/vector_operations.py

```diff
@@ -599,6 +599,71 @@ def _cartesian_z(field, data):
                 display_field=True,
             )
 
+    if registry.ds.geometry == "spherical":
```

Comment:
I really like the usage of the cylindrical components inside our spherical data.